automaxlair/da_controller.py

```diff
@@ -949,26 +913,11 @@ def record_ball_use(self) -> None:
             self.log(
                 f'{ball_to_use} number not initialized before use.', 'ERROR')
 
-        # DEPRECATED: manually track each ball number until the new method is
-        # shown to work.
-        if self.base_ball == self.legendary_ball:
```

Comment:
Check the code immediately above for the ball accounting—it's not highlighted as a change because it was present before but unused. The ball numbers are now stored in a dict.